### PR TITLE
Add speech beta samples

### PR DESCRIPTION
**Pull request by nirupa-kumar**
Diarization,Multi-channel, Multi-language and Word level confidence
@nnegrey Please review

### CODE REVIEW

File: speech/cloud-client/pom.xml [hunk 1]
```diff
@@ -40,7 +40,7 @@
     <dependency>
       <groupId>com.google.cloud</groupId>
       <artifactId>google-cloud-speech</artifactId>
-      <version>0.52.0-alpha</version>
+      <version>0.52.1-alpha-SNAPSHOT</version>
```
Comment (nnegrey) [inline speech/cloud-client/pom.xml]: Update lib version when released.

File: speech/cloud-client/src/main/java/com/example/speech/Recognize.java [hunk 2]
```diff
@@ -779,4 +797,363 @@ public static void transcribeFileWithMetadata(String fileName) throws Exception
...
+
+      // Configure request to enable enhanced models
+      RecognitionConfig config =
+          RecognitionConfig.newBuilder()
+              .setEncoding(AudioEncoding.LINEAR16)
+              .setLanguageCode("en-US")
+              .setSampleRateHertz(8000)
+              .setEnableSpeakerDiarization(true)
+              .setDiarizationSpeakerCount(2)
+              .setEnableAutomaticPunctuation(true)
```
Comment (nnegrey) [inline speech/cloud-client/src/main/java/com/example/speech/Recognize.java]: Let's remove the automatic punctuation for this sample. Unless its needed.

File: speech/cloud-client/src/main/java/com/example/speech/Recognize.java [hunk 3]
```diff
@@ -779,4 +797,363 @@ public static void transcribeFileWithMetadata(String fileName) throws Exception
...
+              .setModel("phone_call")
```
Comment (nnegrey) [inline speech/cloud-client/src/main/java/com/example/speech/Recognize.java]: Are models required here? If not, remove.

File: speech/cloud-client/src/main/java/com/example/speech/Recognize.java [hunk 4]
```diff
@@ -779,4 +797,363 @@ public static void transcribeFileWithMetadata(String fileName) throws Exception
...
+  public static void transcribeDiarization(String fileName) throws Exception {
+    Path path = Paths.get(fileName);
+    byte[] content = Files.readAllBytes(path);
+
+    try (SpeechClient speechClient = SpeechClient.create()) {
+      // Get the contents of the local audio file
+      RecognitionAudio recognitionAudio =
+          RecognitionAudio.newBuilder().setContent(ByteString.copyFrom(content)).build();
+
+      // Configure request to enable enhanced models
```
Comment (nnegrey) [inline speech/cloud-client/src/main/java/com/example/speech/Recognize.java]: // Configure request to use speaker diarization or something along those lines.

File: speech/cloud-client/src/main/java/com/example/speech/Recognize.java [hunk 5]
```diff
@@ -779,4 +797,363 @@ public static void transcribeFileWithMetadata(String fileName) throws Exception
...
+      for (SpeechRecognitionResult result : recognizeResponse.getResultsList()) {
+        // There can be several alternative transcripts for a given chunk of speech. Just
+        // use the first (most likely) one here.
+        SpeechRecognitionAlternative alternative = result.getAlternatives(0);
+        System.out.format("Transcript : %s\n", alternative.getTranscript());
+        System.out.format(
+            "Speaker Tag : %s \n",
+            alternative.getWords((alternative.getWordsCount() - 1)).getSpeakerTag());
+        System.out.format(
+            "Word: %s\n\n", alternative.getWords((alternative.getWordsCount() - 1)).getWord());
```
Comment (nnegrey) [inline speech/cloud-client/src/main/java/com/example/speech/Recognize.java]: Lines: 838-841:
Is this printing out the last speaker and their last word?

Is is possible to maybe do something like: (Not sure what the results look like)
Speaker Tag ###: Hey, how are you?
Speaker Tag ***: I'm doing good. 

Thoughts?

File: speech/cloud-client/src/main/java/com/example/speech/Recognize.java [hunk 6]
```diff
@@ -779,4 +797,363 @@ public static void transcribeFileWithMetadata(String fileName) throws Exception
...
+      // Configure request to enable Speaker diarization
+      RecognitionConfig config =
+          RecognitionConfig.newBuilder()
+              .setEncoding(AudioEncoding.LINEAR16)
+              .setLanguageCode("en-US")
+              .setSampleRateHertz(8000)
+              .setEnableSpeakerDiarization(true)
+              .setDiarizationSpeakerCount(2)
+              .setEnableAutomaticPunctuation(true)
+              .setModel("phone_call")
```
Comment (nnegrey) [inline speech/cloud-client/src/main/java/com/example/speech/Recognize.java]: Same as above, remove setEnableAutomaticPunctuation, and setModel (if possible)

File: speech/cloud-client/src/main/java/com/example/speech/Recognize.java [hunk 7]
```diff
@@ -779,4 +797,363 @@ public static void transcribeFileWithMetadata(String fileName) throws Exception
...
+
+  // [START speech_transcribe_diarization_gcs]
+  /**
+   * Transcribe a remote audio file using speaker diarization.
+   *
+   * @param gcsUri the path to an audio file.
+   */
+  public static void transcribeDiarizationGcs(String gcsUri) throws Exception {
+    try (SpeechClient speechClient = SpeechClient.create()) {
+      // Configure request to enable Speaker diarization
```
Comment (nnegrey) [inline speech/cloud-client/src/main/java/com/example/speech/Recognize.java]: Use this comment above. :)

File: speech/cloud-client/src/main/java/com/example/speech/Recognize.java [hunk 8]
```diff
@@ -779,4 +797,363 @@ public static void transcribeFileWithMetadata(String fileName) throws Exception
...
+
+      // Configure request to enable enhanced models
+      RecognitionConfig config =
+          RecognitionConfig.newBuilder()
+              .setEncoding(AudioEncoding.LINEAR16)
+              .setLanguageCode("en-US")
+              .setSampleRateHertz(44100)
+              .setAudioChannelCount(2)
+              .setEnableSeparateRecognitionPerChannel(true)
+              .setEnableAutomaticPunctuation(true)
```
Comment (nnegrey) [inline speech/cloud-client/src/main/java/com/example/speech/Recognize.java]: Remove setEnableAutomaticPunctuation (if possible)

File: speech/cloud-client/src/main/java/com/example/speech/Recognize.java [hunk 9]
```diff
@@ -779,4 +797,363 @@ public static void transcribeFileWithMetadata(String fileName) throws Exception
...
+  public static void transcribeMultiChannel(String fileName) throws Exception {
+    Path path = Paths.get(fileName);
+    byte[] content = Files.readAllBytes(path);
+
+    try (SpeechClient speechClient = SpeechClient.create()) {
+      // Get the contents of the local audio file
+      RecognitionAudio recognitionAudio =
+          RecognitionAudio.newBuilder().setContent(ByteString.copyFrom(content)).build();
+
+      // Configure request to enable enhanced models
```
Comment (nnegrey) [inline speech/cloud-client/src/main/java/com/example/speech/Recognize.java]: Update comment

File: speech/cloud-client/src/main/java/com/example/speech/Recognize.java [hunk 10]
```diff
@@ -779,4 +797,363 @@ public static void transcribeFileWithMetadata(String fileName) throws Exception
...
+
+      // Configure request to enable multiple channels
+      RecognitionConfig config =
+          RecognitionConfig.newBuilder()
+              .setEncoding(AudioEncoding.LINEAR16)
+              .setLanguageCode("en-US")
+              .setSampleRateHertz(44100)
+              .setAudioChannelCount(2)
+              .setEnableSeparateRecognitionPerChannel(true)
+              .setEnableAutomaticPunctuation(true)
```
Comment (nnegrey) [inline speech/cloud-client/src/main/java/com/example/speech/Recognize.java]: Remove

File: speech/cloud-client/src/main/java/com/example/speech/Recognize.java [hunk 11]
```diff
@@ -779,4 +797,363 @@ public static void transcribeFileWithMetadata(String fileName) throws Exception
...
+   * @param fileName the path to the local audio file
+   */
+  public static void transcribeWordLevelConfidence(String fileName) throws Exception {
+    Path path = Paths.get(fileName);
+    byte[] content = Files.readAllBytes(path);
+
+    try (SpeechClient speechClient = SpeechClient.create()) {
+      RecognitionAudio recognitionAudio =
+          RecognitionAudio.newBuilder().setContent(ByteString.copyFrom(content)).build();
+      // Configure request to enable multiple channels
```
Comment (nnegrey) [inline speech/cloud-client/src/main/java/com/example/speech/Recognize.java]: Update comment

File: speech/cloud-client/src/main/java/com/example/speech/Recognize.java [hunk 12]
```diff
@@ -779,4 +797,363 @@ public static void transcribeFileWithMetadata(String fileName) throws Exception
...
+
+  /**
+   * Transcribe a remote audio file with word level confidence
+   *
+   * @param gcsUri path to the remote audio file
+   */
+  public static void transcribeWordLevelConfidenceGcs(String gcsUri) throws Exception {
+    try (SpeechClient speechClient = SpeechClient.create()) {
+
+      // Configure request to enable multiple channels
```
Comment (nnegrey) [inline speech/cloud-client/src/main/java/com/example/speech/Recognize.java]: Update commment

File: speech/cloud-client/src/main/java/com/example/speech/Recognize.java (hunk 5 above)
Comment (nirupa-kumar) [reply speech/cloud-client/src/main/java/com/example/speech/Recognize.java]: It is printing out the last speaker and last word. The words array contains the entire transcript up until that point. 
Speaker Tag ###: Hey, how are you? : Definitely makes more sense. Will switch it out.

File: speech/cloud-client/src/main/java/com/example/speech/Recognize.java [hunk 13]
```diff
@@ -362,15 +380,15 @@ public static void streamingRecognizeFile(String fileName) throws Exception, IOE
...
-      RecognitionConfig recConfig = RecognitionConfig.newBuilder()
-          .setEncoding(AudioEncoding.LINEAR16)
-          .setLanguageCode("en-US")
-          .setSampleRateHertz(16000)
-          .setModel("default")
-          .build();
-      StreamingRecognitionConfig config = StreamingRecognitionConfig.newBuilder()
-          .setConfig(recConfig)
-          .build();
+      RecognitionConfig recConfig =
```
Comment (jerjou) [inline speech/cloud-client/src/main/java/com/example/speech/Recognize.java]: It'd be helpful if you could do these formatting changes in a separate PR, so that there isn't this giant diff of unrelated changes for reviewers to review..

File: speech/cloud-client/src/main/java/com/example/speech/Recognize.java (hunk 2 above)
Comment (jerjou) [reply speech/cloud-client/src/main/java/com/example/speech/Recognize.java]: +1 Especially in a sample, it should only have the minimum needed to get the feature you're demonstrating working.

File: speech/cloud-client/src/main/java/com/example/speech/Recognize.java (hunk 5 above)
Comment (jerjou) [reply speech/cloud-client/src/main/java/com/example/speech/Recognize.java]: Can you add this as a comment? ie the explanation of why you're getting the last word of the alternative instead of, say, all the words or the first word.

File: speech/cloud-client/src/main/java/com/example/speech/Recognize.java (hunk 13 above)
Comment (nirupa-kumar) [reply speech/cloud-client/src/main/java/com/example/speech/Recognize.java]: Sorry, given my access issues, I'm not sure how it may affect creating a separate PR. Putting everything together in one for now.

**Comment by nirupa-kumar** [comment]
All changes done. Please let me know if this is good to merge.

File: speech/cloud-client/src/main/java/com/example/speech/Recognize.java [hunk 14]
```diff
@@ -779,4 +797,355 @@ public static void transcribeFileWithMetadata(String fileName) throws Exception
...
+
+      // Print out the results
+      for (SpeechRecognitionResult result : recognizeResponse.getResultsList()) {
+        // There can be several alternative transcripts for a given chunk of speech. Just
+        // use the first (most likely) one here.
+        SpeechRecognitionAlternative alternative = result.getAlternatives(0);
+        System.out.format("Transcript : %s\n", alternative.getTranscript());
+        // The words array contains the entire transcript up until that point.
+        //Referencing the last spoken word to get the associated Speaker tag
+        System.out.format("Speaker Tag %s:%s\n",
```
Comment (nnegrey) [inline speech/cloud-client/src/main/java/com/example/speech/Recognize.java]: Add a space between %s:%s --> %s: %s

File: speech/cloud-client/src/main/java/com/example/speech/Recognize.java (hunk 9 above)
Comment (nnegrey) [reply speech/cloud-client/src/main/java/com/example/speech/Recognize.java]: // Configure request to enable multiple channels

**Comment by nirupa-kumar** [comment]
I signed it!

**Comment by nirupa-kumar** [comment]
Checking if this triggers the CLA bot

**Comment by nirupa-kumar** [comment]
@nnegrey - Updated the client library too. Please let me know if this is good to merge

**Comment by nnegrey** [comment]
Per an email from OSPO, nirupa-kumar has a signed CLA.
